cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/ReplicationService.java

```diff
@@ -0,0 +1,323 @@
+     */
+    public static class SimpleReplicationCompleteListener implements ReplicationCompleteListener {
+        @Override
+        public void allReplicationsComplete() {
+        }
+
+        @Override
+        public void replicationComplete(int id) {
+        }
+
+        @Override
+        public void replicationErrored(int id) {
+        }
+    }
+
+    // A binder to allow components running in the same process to bind to this Service.
+    public class LocalBinder extends Binder {
+        public ReplicationService getService() {
+            return ReplicationService.this;
+        }
+    }
+
+    // Handler that receives messages from the thread.
+    protected class ServiceHandler extends Handler {
+        public ServiceHandler(Looper looper) {
+            super(looper);
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            // Process the commands passed in msg.arg2.
+            switch (msg.arg2) {
+                case COMMAND_START_REPLICATION:
+                    startReplications();
+                    notifyTestOperationComplete();
+                    break;
+                case COMMAND_STOP_REPLICATION:
+                    stopReplications();
+                    notifyTestOperationComplete();
+                    break;
+            }
+
+            // Get the Intent used to start the service and release the WakeLock if there is one.
+            // Calling completeWakefulIntent is safe even if there is no wakelock held.
+            Intent intent = msg.getData().getParcelable(EXTRA_INTENT);
+            WakefulBroadcastReceiver.completeWakefulIntent(intent);
+        }
+    }
+
+    protected Handler getHandler(Looper looper) {
+        return new ServiceHandler(looper);
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        // Create a background priority thread to so we don't block the process's main thread.
+        HandlerThread thread = new HandlerThread("ServiceStartArguments",
+            android.os.Process.THREAD_PRIORITY_BACKGROUND);
+        thread.start();
+
+        // Get the HandlerThread's Looper and use it for our Handler.
+        Looper serviceLooper = thread.getLooper();
+        mServiceHandler = getHandler(serviceLooper);
+
+        Replicator[] replicators = getReplicators(getApplicationContext());
+
+        if (replicators != null && replicators.length > 0) {
+            if (mReplicationPolicyManager == null) {
+                mReplicationPolicyManager = new ReplicationPolicyManager() {
+                    @Override
+                    public void start() {
+                    }
+
+                    @Override
+                    public void stop() {
+                    }
+                };
+            }
+            mReplicationPolicyManager.addReplicators(replicators);
+            mReplicationPolicyManager.setReplicationsCompletedListener(this);
+            mReplicationPolicyManager.start();
```

Comment:
Could you explain why this conditionless policy manager is created, and how it interacts with the `setReplicationPolicyManager` below? (e.g., should `setReplicationPolicyManager` call `stop` on any existing policy manager before re-assigning?)